src/core_plugins/kibana/public/dashboard/directives/dashboard_panel.js

```diff
@@ -1,15 +1,19 @@
 import _ from 'lodash';
 import 'ui/visualize';
 import 'ui/doc_table';
-import { loadPanelProvider } from 'plugins/kibana/dashboard/components/panel/lib/load_panel';
+import 'plugins/kibana/dashboard/services/get_object_loaders_for_dashboard.factory';
 import FilterManagerProvider from 'ui/filter_manager';
 import uiModules from 'ui/modules';
 import panelTemplate from 'plugins/kibana/dashboard/components/panel/panel.html';
+import { getPersistedStateId } from 'plugins/kibana/dashboard/components/panel/lib/panel_state';
+import { loadSavedObject } from 'plugins/kibana/dashboard/components/panel/lib/load_saved_object';
 
 uiModules
 .get('app/dashboard')
-.directive('dashboardPanel', function (savedVisualizations, savedSearches, Notifier, Private, $injector) {
-  const loadPanel = Private(loadPanelProvider);
+.directive('dashboardPanel', function ($injector) {
+  const Private = $injector.get('Private');
```

Comment:
Okay folks, I reverted to the old variation.  Discussed with @cjcenizal and @w33ble and decided to move the debate around the new $injector style to a separate issue and pull out of here, in order to stay consistent with the current code, until the new pattern is embraced as a whole by the team.